config/config.go

```diff
@@ -100,7 +101,7 @@ func (cfg *Config) overrideHeader(newHeader http.Header) {
 // WithURL sets the current Config to the parsed *url.URL from rawURL
 // and returns it. Any errors is discarded as a Config can be invalid
 // until Config.Validate is called. The url is guaranteed not to be nil.
-func (cfg Config) WithURL(rawURL string) Config {
+func (cfg Global) WithURL(rawURL string) Global {
```

Comment:
As `Config.Request` is now a standalone (kinda), would it be better if `WithURL` is its method ? 

 ```suggestion
func (r Request) WithURL(rawURL string) Request {
```

And in main:

```go
	cliCfg := config.Global{
		Request: config.Request{
			Header: header,
		}.WithURL(uri),
		Runner: config.Runner{
			// ...
		},
	}
```